### PR TITLE
compat DFG 25 AMP 9

### PR DESCRIPTION
**Pull request by dehann**
Replaces:
- #758 
- #760

### CODE REVIEW

**Comment by Affie** [comment]
I didn't see this so merged the compat PRs because my environment was broken. I'll resolve the conflicts.

**Comment by Affie** [comment]
> oops, unexpected test failure in IMU delta time calculation:

This is unexpected, I thought I fixed the IMU timing issues. I don't think the fix is to increase the tolerance, the check is there to show that something is wrong. It's fine for now, but we should look at why the time is out.

File: src/factors/Inertial/IMUDeltaFactor.jl [hunk 1]
```diff
@@ -355,7 +355,7 @@ function (cf::CalcFactor{<:IMUDeltaFactor})(
     Xhat = log_lie(M, Manifolds.compose(M, inv(M, Δi), Δhat))
 
     Xc_hat = vee(M, Xhat)
-    @assert isapprox(Δi.x[4], Δhat.x[4], atol=1e-6) "Time descrepancy in IMUDeltaFactor: Δt = $(Xc_hat[10])"
+    @assert isapprox(Δi.x[4], Δhat.x[4], atol=2e-6) "Time descrepancy in IMUDeltaFactor: Δt = $(Xc_hat[10]), $(Δi.x[4]), $(Δhat.x[4])"
```
Comment (Affie) [inline src/factors/Inertial/IMUDeltaFactor.jl:358]:
```suggestion
    @assert isapprox(Δi.x[4], Δhat.x[4], atol=1e-6) "Time descrepancy in IMUDeltaFactor: Δt = $(Xc_hat[10]), $(Δi.x[4]), $(Δhat.x[4])"
```

**Comment by Affie** [comment]
IMU factor tests should be fixed now. Debugging test slipped through the cracks.

**Comment by dehann** [comment]
Thanks!

**Comment by dehann** [comment]
So there is a stochastic issue with testInertialDynamic (reduced stack trace below).  I've been struggling to figure out what's causing it and been trying a few things.  Here is the only things I can say for sure so far:
- Sometimes the tests pass, sometimes they fail.
- Sometimes the failures seem to come in batches or be related to how much time elapsed since the last test, but there is barely something distinctive.  This lead me to think it has something to do with the random number generator.
- If you approxConv many points (i.e. `getSolverParams(fg).N = 1000`) the tests fails every time at a different sample number.  Converse, if you reduce to say `N=40` then the approxConv tends to succeed more often.  Why some points are bad is still unknown.

**Comment by dehann** [comment]
Okay, saveDFG did not work on testInertialDynamic's graph but I was able to capture at least the variable values from two contrasting graphs (good vs bad):

[inerdyn_N30_vars.zip](https://github.com/user-attachments/files/17954397/inerdyn_N30_vars.zip)

**Comment by dehann** [comment]
now starting to think the ODE solution of INS is wondering away from proper rotation matrix and this is inducing errors during propagation.  Not sure how that is manifesting into the extrapolation issue yet, but suspect the solution is somehow wandering off the "INS manifold"

**Comment by dehann** [comment]
deferring since there seems to be a 40% success rate with the test and the issue seems to be something more intricate.

**Pull request merged**